### PR TITLE
[candi] Fix bash word splitting

### LINKED ISSUE #7387: [candi][bashible] Add support of registry cr.yandex
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Version

v1.56.9

### Expected Behavior

When we use cr.yandex (Container Registry by Yandex https://cloud.yandex.com/en-ru/services/container-registry) as registry for deckhouse all works fine

### Actual Behavior

When we try create instance we have problem with bashible like this
```
+++ command -v d8-curl
+++ curl --retry 3 -fsSL -u 'json_key:{"id":"aj**********ga","service_account_id":"aj*********u4","created_at":"2020-12-02T12:45:11.631357Z","key_algorithm":"RSA_2048","public_key":"-----BEGIN' PUBLIC 'KEY-----\nTWFueSBzeW1ib2xzIG9mIGJhc2U2NCBlbmNvZGluZyBwdWJsaWMga2V5\n-----END' PUBLIC 'KEY-----\n","private_key":"-----BEGIN' PRIVATE 'KEY-----\=\n-----END' PRIVATE 'KEY-----\n"}' 'https://cr.yandex/v2/token/?service=cr.yandex&scope=repository:cr******3ch/deckhouse/fe:pull'
curl: (6) Could not resolve host: PUBLIC
curl: (6) Could not resolve host: PUBLIC
curl: (6) Could not resolve host: PUBLIC
curl: (6) Could not resolve host: PUBLIC
```

### Steps To Reproduce

1. Create `Yandex Cloud Container Registry`
2. Change registry by `change-registry` helper of existing cluster to cr.yandex
3. Check works of recreate instances

### Additional Information

https://cloud.yandex.ru/ru/docs/container-registry/operations/authentication#sa-json
Also, need to check, that `change-registry` helper works correctly with authenticate data of yandex

### Logs

_No response_

### PR DESCRIPTION
**Pull request by asviel**
## Description

Fix bash word splitting.

## Why do we need it, and what problem does it solve?

Closes #7387 

## Why do we need it in the patch release (if we do)?

Yes, we do. This affects the client.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: candi
type: fix
summary: Fix bash word splitting.
impact_level: default
```

### CODE REVIEW

**Review by RomanenkoDenys — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by asviel** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.25 7815513259 7815513259-1-con-1-25 /sys/deckhouse-oss/install:pr7410 cloud-user@178.154.207.189

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: candi/bashible/bashbooster/50_deckhouse_registrypackages.sh [hunk 1]
```diff
@@ -64,19 +64,18 @@ bb-rp-is-fetched?() {
   local AUTH_REALM=""
   local AUTH_SERVICE=""
 
-  if [[ -n ${REGISTRY_AUTH} ]]; then
-    AUTH="-u ${REGISTRY_AUTH}"
+  if [[ -n "${REGISTRY_AUTH}" ]]; then
+    AUTH="yes"
```
Comment (RomanenkoDenys) [inline candi/bashible/bashbooster/50_deckhouse_registrypackages.sh:68]: this var should be local to avoid misconfigurations between runs

**Review by RomanenkoDenys — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [7422](https://github.com/deckhouse/deckhouse/pull/7422) to the branch [release-1.57](https://github.com/deckhouse/deckhouse/tree/release-1.57) successful!

**Pull request merged**